### PR TITLE
Fix some issues with Ignitor

### PR DESCRIPTION
**Pull request by pranavkm**
* Avoid null ref when FormatException is not set
* Handle errors when the client is canceled but an operation is in progress

### CODE REVIEW

File: src/Components/Ignitor/src/CancellableOperation.cs [hunk 1]
```diff
@@ -5,37 +5,50 @@
 using System.Threading;
 using System.Threading.Tasks;
 
+#nullable enable
 namespace Ignitor
 {
     internal class CancellableOperation<TResult>
     {
-        public CancellableOperation(TimeSpan? timeout)
+        public CancellableOperation(TimeSpan? timeout, CancellationToken cancellationToken)
```
Comment (pranavkm) [inline src/Components/Ignitor/src/CancellableOperation.cs]: The bulk of the interesting changes are localized to this type. Here's what happens:

1) We start a `PrepareForNextBatch`
2) In the meanwhile, `BlazorClient.Cancel` is invoked (the server may or may not be dead at this point)
3) `CancellableOperation` knows nothing about this and eventually times out.

This change associates the `BlazorClient.Cancel` with the operation and no-ops if the client has been cancelled

File: src/Components/Ignitor/src/BlazorClient.cs [hunk 2]
```diff
@@ -233,7 +236,7 @@ private async Task<CapturedRenderBatch> WaitForRenderBatch(TimeSpan? timeout = n
                 {
                     return await PrepareForNextBatch(timeout ?? DefaultOperationTimeout);
                 }
-                catch (OperationCanceledException)
+                catch (TimeoutException) when (FormatError != null)
```
Comment (pranavkm) [inline src/Components/Ignitor/src/BlazorClient.cs]: This was a pain to discover: `FomatError` isn't set up in the benchmarks repo. Took me forever to figure out was causing a null ref.

Comment (javiercn) [reply src/Components/Ignitor/src/BlazorClient.cs]: @pranavkm What is format error?

**Review by javiercn — APPROVED** [review]
Great improvements!

**Review by NTaylorMullen — APPROVED** [review]

File: src/Components/Ignitor/src/BlazorClient.cs (hunk 2 above)
Comment (pranavkm) [reply src/Components/Ignitor/src/BlazorClient.cs]: It's a callback used in the test to pretty print the exception + log: https://github.com/aspnet/AspNetCore/blob/master/src/Components/test/E2ETest/ServerExecutionTests/IgnitorTest.cs#L59-L63.

**Pull request merged**